### DISCUSSION TITLE
Install GCC to cross-compile from x86_64 to aarch64

### DISCUSSION THREAD

**Discussion by dEajL3kA** [Homebrew/discussions#3203] [Getting started]
Is it possible to cross-compile to "aarch64" (Apple Silicon) from a "x86_64" Mac with GCC, as provided by Homebrew?

With the Apple clang compiler, we can simply do:
```
$(CC) main.c -o x86_app -target x86_64-apple-darwin
$(CC) main.c -o arm_app -target arm64-apple-darwin
```

...but GCC doesn't seem to support the "-target" option 😞 

On Linux, I'm used to install a special "crossbuild" version of GCC to accomplish this. For example, on Debian I would go with:
```
apt install crossbuild-essential-arm64
```

...which gives me a separate **aarch64-linux-gnu-gcc** binary to target ARM64-based Linux platforms:
```
aarch64-linux-gnu-gcc main.c -o arm_app
`````

Is there a similar package/formula provided by Homebrew? So far, I only found this:
```
aarch64-elf-gcc
```

...but I'm not sure whether this is the right compiler to target Apple Silicon. Actually, "elf" sounds wrong to me!

I think we'd need something like this, but I cannot find it in Homebrew:
```
arm64-apple-darwin-gcc
```

----

Background: I only have a "x86_64" Macbook available, so I have to cross-compile in order to support Apple Silicon for now.

But I found that, at least for the "x86_64" target, GCC produces ***much*** faster code.... (compared to clang)

Any advice would be much appreciated.
Thank you!

**Comment by lzy-0321** [comment]
https://github.com/lzy-0321/COMP3601/blob/main/README.md
Here are my experiences, hope can help you!
